seacrowd/sea_datasets/qed/qed.py

```diff
@@ -0,0 +1,285 @@
+"""
+
+_HOMEPAGE = "https://opus.nlpl.eu/QED/corpus/version/QED"
+
+_LANGUAGES = ["eng", "vie", "tha", "mya", "jav", "ind", "tgl", "zlm"]
+
+_LICENSE = Licenses.OTHERS.value
+
+_LOCAL = False
+
+_FILE = "QED.{}.{}"  # E.g. QED.en-id.id
+
+_URLS = "https://object.pouta.csc.fi/OPUS-QED/v2.0a/moses/{}.txt.zip"
+
+_SUPPORTED_TASKS = [Tasks.MACHINE_TRANSLATION, Tasks.SELF_SUPERVISED_PRETRAINING]
+
+_SOURCE_VERSION = "2.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class QEDDataset(datasets.GeneratorBasedBuilder):
+    """QED - The QCRI Educational Domain Corpus (formerly QCRI AMARA Corpus) is an open multilingual collection of subtitles for educational videos and lectures collaboratively transcribed and translated over the AMARA web-based platform. It's developed by Qatar Computing Research Institute, Arabic Language Technologies Group. Along with English, it covers multiple SEA languages, such as vi (Vietnamese), my (Burnmese), jv (Javanese), id (Indonesia), th (Thai), tl (Tagalog), ms (Malaysia)."""
+
+    SEACROWD_SCHEMA = TASK_TO_SCHEMA[_SUPPORTED_TASKS[0]].lower()
+
+    LANG_PAIRS = [
+        ("en", "vi"),
+        ("en", "th"),
+        ("en", "my"),
+        ("en", "jv"),
+        ("en", "id"),
+        ("en", "tl"),
+        ("en", "ms"),
+        ("th", "vi"),
+        ("th", "my"),
+        ("th", "jv"),
+        ("th", "tl"),
+        ("my", "tl"),
+        ("my", "vi"),
+        ("jv", "vi"),
+        ("jv", "my"),
+        ("jv", "tl"),
+        ("jv", "ms"),
+        ("id", "jv"),
+        ("id", "th"),
+        ("id", "vi"),
+        ("id", "my"),
+        ("id", "tl"),
+        ("id", "ms"),
+        ("tl", "vi"),
+        ("ms", "tl"),
+        ("ms", "th"),
+        ("ms", "vi"),
+        ("ms", "my"),
+    ]
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_{lang1}-{lang2}_source",
+            version=datasets.Version(_SOURCE_VERSION),
+            description=f"{_DATASETNAME} source schema for translation from {lang1} to {lang2}",
+            schema="source",
+            subset_id=f"{_DATASETNAME}_{lang1}-{lang2}",
+        )
+        for lang1, lang2 in LANG_PAIRS
+    ] + [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_{lang1}-{lang2}_{lang1}_source",
+            version=datasets.Version(_SEACROWD_VERSION),
+            description=f"{_DATASETNAME} source schema {lang1} for translation from {lang1} to {lang2}",
+            schema=f"source",
+            subset_id=f"{_DATASETNAME}_{lang1}-{lang2}_{lang1}",
+        )
+        for lang1, lang2 in LANG_PAIRS
+    ] + [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_{lang1}-{lang2}_{lang2}_source",
+            version=datasets.Version(_SEACROWD_VERSION),
+            description=f"{_DATASETNAME} source schema {lang2} for translation from {lang1} to {lang2}",
+            schema=f"source",
+            subset_id=f"{_DATASETNAME}_{lang1}-{lang2}_{lang2}",
+        )
+        for lang1, lang2 in LANG_PAIRS
```

Comment:
So a particular language `X` will have different texts as a source and target language, no? I was thinking that maybe it's better to combine them instead into one since we're just going to use it for language modeling? So something like `{_DATASET_NAME}_{lang}_source`. Wdyt @elyanah-aco @holylovenia @SamuelCahyawijaya @sabilmakbar ?

Reply:
@ljvmiranda921 I think it's fine for the `source` schemas to be constructed in the exact manner of the original data, which is what @patrickamadeus did. I agree it doesn't use the most intuitive composition though.

Also, is this dataset parallel? I don't know if `{lang1}-{lang2}_{lang1}` and `{lang3}-{lang1}_{lang1}` contain the same data instances or not.

I think we can keep the `source` schemas simple and be more task-driven with the `seacrowd` schemas.

cc: @elyanah-aco @sabilmakbar @SamuelCahyawijaya

Reply:
Hi! Thanks for all of the reviews!

To justify my initial approach, for `ssp` task, I thought the bilingual task contains different article for `lang1` in case of `lang1-lang2_lang1` vs `lang3-lang1_lang1`. 

Turns out I missed the monolingual download section and I have verified that both cases share the same `lang1` source.

<img width="1334" alt="image" src="https://github.com/SEACrowd/seacrowd-datahub/assets/65403659/d7c909ea-b7e2-49e3-8972-1764802c016f">

Will update this soon!